yt_dlp/extractor/swearnet.py

```diff
@@ -58,22 +32,17 @@ def _real_extract(self, url):
                 self.raise_login_required()
             raise
 
-        json_data = self._download_json(
-            f'https://play.vidyard.com/player/{external_id}.json', display_id)['payload']['chapters'][0]
-
-        formats, subtitles = self._get_formats_and_subtitle(json_data['sources'], display_id)
-        self._merge_subtitles(self._get_direct_subtitle(json_data.get('captions')), target=subtitles)
-
-        return {
-            'id': str(json_data['videoId']),
-            'title': json_data.get('name') or self._html_search_meta(['og:title', 'twitter:title'], webpage),
-            'description': (json_data.get('description')
-                            or self._html_search_meta(['og:description', 'twitter:description'], webpage)),
-            'duration': int_or_none(json_data.get('seconds')),
-            'formats': formats,
-            'subtitles': subtitles,
+        video_json = self._fetch_video_json(external_id, display_id)
+        video_info = {
+            **self._process_video_json(video_json['chapters'][0], display_id),
             'season_number': int_or_none(season_number),
             'episode_number': int_or_none(episode_number),
-            'thumbnails': [{'url': thumbnail_url}
-                           for thumbnail_url in traverse_obj(json_data, ('thumbnailUrls', ...))],
         }
+
+        if video_info['title'] is None:
```

Comment:
Maybe `title` must have been set but this looks safer:
```suggestion
        if not video_info.get('title'):
```